sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.institutions.controller.getOfferingSummary.e2e-spec.ts

```diff
@@ -0,0 +1,336 @@
+  getAuthorizedLocation,
+} from "../../../../testHelpers";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  OfferingDeliveryOptions,
+  createFakeEducationProgram,
+  createFakeUser,
+} from "@sims/test-utils";
+import {
+  OfferingIntensity,
+  OfferingStatus,
+  InstitutionUserTypes,
+  EducationProgramOffering,
+} from "@sims/sims-db";
+
+describe("EducationProgramOfferingInstitutionsController(e2e)-getOfferingSummary", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let fullTimeOffering: EducationProgramOffering;
+  let partTimeOffering: EducationProgramOffering;
+  let baseEndpoint: string;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    const sharedUser = createFakeUser();
+    await db.user.save(sharedUser);
+
+    // Get College C institution and location.
+    const { institution } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const collegeC = institution;
+    const collegeCLocation = await getAuthorizedLocation(
+      db,
+      InstitutionTokenTypes.CollegeCUser,
+      InstitutionUserTypes.admin,
+    );
+
+    const program = createFakeEducationProgram({
+      institution: collegeC,
+      auditUser: sharedUser,
+    });
+    await db.educationProgram.save(program);
+
+    // Create test offerings with different intensities and dates.
+    fullTimeOffering = createFakeEducationProgramOffering(
+      program,
+      collegeCLocation,
+    );
+
+    fullTimeOffering.offeringIntensity = OfferingIntensity.fullTime;
+    fullTimeOffering.studyStartDate = "2024-08-01";
+    fullTimeOffering.studyEndDate = "2024-12-31";
+    fullTimeOffering.offeringStatus = OfferingStatus.Approved;
+    fullTimeOffering.offeringDelivered = OfferingDeliveryOptions.Online;
+    fullTimeOffering.name = "Full-Time Test Offering";
+
+    await db.educationProgramOffering.save(fullTimeOffering);
+
+    partTimeOffering = createFakeEducationProgramOffering(
+      program,
+      collegeCLocation,
+    );
+
+    partTimeOffering.offeringIntensity = OfferingIntensity.partTime;
+    partTimeOffering.studyStartDate = "2025-01-01";
+    partTimeOffering.studyEndDate = "2025-05-31";
+    partTimeOffering.offeringStatus = OfferingStatus.Approved;
+    partTimeOffering.offeringDelivered = OfferingDeliveryOptions.Blended;
+    partTimeOffering.name = "Part-Time Test Offering";
+    await db.educationProgramOffering.save(partTimeOffering);
+
+    baseEndpoint = `/institutions/education-program-offering/location/${collegeCLocation.id}/education-program/${program.id}?page=0&pageLimit=10`;
+  });
+
+  it("Should return all offerings when no filters are applied.", async () => {
+    // Arrange
+    const token = await getInstitutionToken(InstitutionTokenTypes.CollegeCUser);
+
+    const endpoint = baseEndpoint;
+
+    // Act & Assert
```

Comment:
Minor. For consistency we can use `Act/Assert` as it is followed all over the application.